eis_toolkit/vector_processing/rasterize_vector.py

```diff
@@ -0,0 +1,125 @@
+        )
+
+    if base_raster_profile is not None and not isinstance(base_raster_profile, (profiles.Profile, dict)):
+        raise exceptions.InvalidParameterValueException(
+            f"Expected base_raster_profile ({type(base_raster_profile)}) to be dict or rasterio.profiles.Profile."
+        )
+
+    if buffer_value is not None:
+        geodataframe = geodataframe.copy()
+        geodataframe["geometry"] = geodataframe["geometry"].apply(lambda geom: geom.buffer(buffer_value))
+
+    return _rasterize_vector(
+        geodataframe=geodataframe,
+        value_column=value_column,
+        default_value=default_value,
+        fill_value=fill_value,
+        base_raster_profile=base_raster_profile,
+        resolution=resolution,
+    )
+
+
+def _transform_from_geometries(
+    geodataframe: gpd.GeoDataFrame, resolution: float
+) -> Tuple[float, float, transform.Affine]:
+    """Determine transform from the input geometries.
+
+    Returns:
+        width (float): Width of the raster
+        height (float): Height of the raster
+        out_transform (rasterio.transform.Affine): Affine transform of the output
+    """
+    min_x, min_y, max_x, max_y = geodataframe.total_bounds
+    width = (max_x - min_x) / resolution
+    height = (max_y - min_y) / resolution
+
+    out_transform = transform.from_bounds(min_x, min_y, max_x, max_y, width=width, height=height)
+    return width, height, out_transform
+
+
+def _rasterize_vector(
+    geodataframe: gpd.GeoDataFrame,
+    value_column: Optional[str],
+    default_value: float,
+    fill_value: float,
+    base_raster_profile: Optional[profiles.Profile],
+    resolution: float,
+) -> Tuple[np.ndarray, dict]:
+    # rasterio.features.rasterize expects a shapes parameter which is
+    # an iterable of tuples where the first value is a geometry and
+    # the other a value for the geometry
+    # Alternatively, if there are not values for each geometry,
+    # an iterable of geometries can be passed
+    geometries = geodataframe["geometry"].values
+    values = geodataframe[value_column].values if value_column is not None else None
+    geometry_value_pairs = list(geometries) if values is None else list(zip(geometries, values))
+
+    if base_raster_profile is None:
+        width, height, out_transform = _transform_from_geometries(geodataframe=geodataframe, resolution=resolution)
+    else:
+        width, height, out_transform = (
+            base_raster_profile["width"],
+            base_raster_profile["height"],
+            base_raster_profile["transform"],
+        )
+
+    out_raster_array = features.rasterize(
+        shapes=geometry_value_pairs,
+        # fill and default_value can be floats even though typing claims otherwise
+        fill=fill_value,
+        default_value=default_value,
+        transform=out_transform,
+        out_shape=(round(height), round(width)),
+    )
+    return out_raster_array, dict(transform=out_transform, height=height, width=width)
```

Comment:
I wonder should we return more metadata here? We could add at least "count", "driver", "crs" and "dtype".

Reply:
I believe the question is if the output raster file needs to be composeable from the output dictionary/profile. If we require the individual functions to collect the needed keys and values into a working `rasterio.profiles.Profile` or `dict`, we should have an utility function to create it and check the output rather than trusting that each module properly returns the right keys and values. 

I only included `transform` and the dimensions as they are the only part that is only known within the function. `crs` can be extracted from the `geodataframe`, `count` is always 1, `driver` depends on what file we want to create and `dtype` can be extracted from the `out_raster_array`.

Reply:
Ok, perhaps this is best to leave as it is now. We can change it later if needed, or create a utility function as you mentioned.